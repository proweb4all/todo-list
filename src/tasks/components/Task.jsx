import React from 'react';
import classNames from 'classnames';

const Task = ({id, text, done, onChange, onDelete}) => {
  const taskClasses = classNames('list-item', {'list-item_done': done})
  return (
    <li className={taskClasses}>
      <input 
        type="checkbox" 
        className='list-item__checkbox' 
        defaultChecked={done} 
        onChange={() => onChange(id)}
      />
      <span className='list-item__text'>{text}</span>
      <button 
        className='list-item__delete-btn'         
        onClick={() => onDelete(id)}
      >
      </button>
    </li>
  )
}

export default Task;