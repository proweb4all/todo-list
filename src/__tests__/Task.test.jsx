import React from 'react'
import Task from '../Task'
import {shallow} from 'enzyme'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

describe('<Task />', () => {
  it('should display done Task', () => {
    const props = {
      text: 'Task 1',
      id: 'some-id-1',
      done: true,
      onChange: jest.fn(),
      onDelete: jest.fn(),
    }
    const wrappedComponent = shallow(<Task {...props} />)
    expect(wrappedComponent).toMatchSnapshot()
  })
  it('should display undone Task', () => {
    const props = {
      text: 'Task 1',
      id: 'some-id-1',
      done: false,
      onChange: jest.fn(),
      onDelete: jest.fn(),
    }
    const wrappedComponent = shallow(<Task {...props} />)
    expect(wrappedComponent).toMatchSnapshot()
  })
  it('should update task on checkbox checked', () => {
    const props = {
      text: 'Task 1',
      id: 'some-id-1',
      done: false,
      onChange: jest.fn(),
      onDelete: jest.fn(),
    }
    const wrappedComponent = shallow(<Task {...props} />)
    wrappedComponent.find('.list-item__checkbox').simulate('change')
    expect(props.onChange).toBeCalledWith('some-id-1')
  })
  it('should delete task', () => {
    const props = {
      text: 'Task 1',
      id: 'some-id-1',
      done: false,
      onChange: jest.fn(),
      onDelete: jest.fn(),
    }
    const wrappedComponent = shallow(<Task {...props} />)
    wrappedComponent.find('.list-item__delete-btn').simulate('click')
    expect(props.onDelete).toBeCalledWith('some-id-1')
  })
})