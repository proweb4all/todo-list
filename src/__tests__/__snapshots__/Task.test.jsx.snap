// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Task /> should display done Task 1`] = `ShallowWrapper {}`;

exports[`<Task /> should display undone Task 1`] = `ShallowWrapper {}`;
